extensions/wikia/WikiFactory/api/MarkWikiAsClosedController.class.php

```diff
@@ -21,18 +23,40 @@ public function markWikiAsClosed() {
 		$wikiId = $request->getVal( self::WIKI_ID );
 		$reason = $request->getVal( self::REASON );
 		$userId = $request->getVal( self::USER_ID );
+		$fandomCreatorCommunityId = $request->getVal( self::FANDOM_CREATOR_COMMUNITY_ID );
 
 		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
 
-		if ( !is_numeric( $wikiId ) || empty( $reason ) || !is_numeric( $userId)) {
+		if ( !empty( $fandomCreatorCommunityId ) && !is_numeric( $fandomCreatorCommunityId ) ) {
+			$this->response->setCode( 400 );
+			$this->info('invalid  fandomCreatorCommunityId parameter in request');
+		} else if ( !is_numeric( $wikiId ) || empty( $reason ) || !is_numeric( $userId ) ) {
```

Comment:
I'd prefer early returns instead of if/else if chaining for validation—makes the flow a bit easier to follow IMO

Reply:
Agreed, done.
Refactored the whole bit, for cleanup.